atlas/protocols.py

```diff
@@ -82,46 +95,44 @@ def generateAtlasStep(self, movieDict):
         # Movie is a dictionary, convert it to a proper movie object
         movie = Movie()
         movie.setAcquisition(Acquisition())
-
         movie.setAttributesFromDict(movieDict, setBasic=True,
                                     ignoreMissing=True)
 
         # Generate the output
-        outputMovies = self._getOutputSet()
-
-        newMovie = Movie()
-        newMovie.copy(movie)
-        self._addAtlasInfo(newMovie)
-        outputMovies.append(newMovie)
-        outputMovies.write()
+        outputLocations = self._getOutputSet()
 
-        self._store()
+        al = self._getAtlasInfo(movie)
+        if al is not None:
+            outputLocations.append(al)
+            outputLocations.write()
+            self._store()
 
-    def _addAtlasInfo(self, movie):
+    def _getAtlasInfo(self, movie):
 
         # Get the parser
-        importProtocol = self.importProtocol.get()
-
-        parser = EPUParser(importProtocol.filesPath.get())
-
-        # create the atlas
-        atlasLoc = AtlasLocation()
+        parser = self._getParser()
 
         try:
-            parser.decorateMovie(importProtocol, movie, atlasLoc)
+            return parser.getAtlasLocation(movie)
         except Exception as e:
-            print ("EPU parser can't add atlas location for %s. Error: %s" % (movie.getMicName(), e))
+            print("EPU parser can't add atlas location for %s. Error: %s" % (movie.getMicName(), e))
+
+    def _createSetOfAtlasLocation(self, suffix=''):
 
-        setAtlasToMovie(movie, atlasLoc)
+        # Intentionally accessing a private createSet method
+        return self._EMProtocol__createSet(SetOfAtlasLocations,
```

Comment:
In scipion3, Sets have a "factory" classmethod:

SetOfAtlasLocations.create(same params)

Should work.